base/record_replay.h

```diff
@@ -175,7 +175,8 @@ struct CompareMemberByRecordReplayId {
     if (IsRecordingOrReplaying("pointer-ids")) {
       int ida = a.Get()->RecordReplayId();
       int idb = b.Get()->RecordReplayId();
-      CHECK(ida && idb);
+      // Allow for zero ids when we may have diverged from the recording.
+      CHECK(HasDivergedFromRecording() || (ida && idb));
```

Comment:
We should still produce a total order when the record/replay IDs are missing, otherwise sorting won't work.  If we sort all objects with IDs before objects without IDs, and then use the pointer based sorting for objects missing their IDs, then things should work.

Reply:
That was my original iteration of this patch, then I considered that sorting on 0 would simply yield a no-op sort (assuming a stable sorting algorithm), with elements with 0 id occurring first in the list.

Also, we currently use 0 ids when we're not recording/replaying, which would create this same behaviour.  So it feels like extra code to force semantic guarantees that we're not are not enforced in other circumstances.

I'll go make the change requested, but lemme know if there's anything wrong with the above reasoning.

Reply:
Hmm, when not recording/replaying we won't be using the IDs, and will sort based on the address.  A no-op sort won't always be correct to do, it will break std::set and other containers if this is used as the comparator, and there are places in chromium that expect to be able to get a total ordering on certain kinds of objects (mojo ports are one example).

Reply:
> Hmm, when not recording/replaying we won't be using the IDs

Right, didn't see that conditional within the comparator where it falls back to pointer compare if we're not recording/replaying.  Got it.